### PR TITLE
Add grpc_status_number as a default access log field

### PR DESCRIPTION
**Pull request by rajatvig**
Fixes https://github.com/projectcontour/contour/issues/4484

Signed-off-by: Rajat Vig <rvig@etsy.com>

### CODE REVIEW

**Comment by rajatvig** [comment]
Could this be reviewed please?

File: changelogs/unreleased/4880-rajatvig-small.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Add `grpc_status_code` to the log message field
```
Comment (skriss) [inline changelogs/unreleased/4880-rajatvig-small.md:1]:
```suggestion
Add `grpc_status_number` to the default JSON access log fields
```

**Review by skriss — CHANGES_REQUESTED** [review]
One minor comment on the changelog, otherwise LGTM.

cc @sunjayBhatia

File: apis/projectcontour/v1alpha1/accesslog.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ var DefaultAccessLogJSONFields = AccessLogJSONFields([]string{
 	"user_agent",
 	"x_forwarded_for",
 	"grpc_status",
+	"grpc_status_number",
```
Comment (sunjayBhatia) [inline apis/projectcontour/v1alpha1/accesslog.go:46]: huh seems like we already added this to `envoySimpleOperators` in this file

Comment (sunjayBhatia) [reply apis/projectcontour/v1alpha1/accesslog.go:46]: (the access log format operator version)

**Review by sunjayBhatia — APPROVED** [review]
might want to be more judicious about the default access log fields in the future, but seems good to add this one

**Comment by sunjayBhatia** [comment]
> might want to be more judicious about the default access log fields in the future, but seems good to add this one

lgtm minus @skriss' comments I should say

**Comment by rajatvig** [comment]
> > might want to be more judicious about the default access log fields in the future, but seems good to add this one
> 
> lgtm minus @skriss' comments I should say

Addressed the comments. Agree, the default access log fields should only be judiciously added.

**Review by skriss — APPROVED** [review]

**Comment by skriss** [comment]
Thanks @rajatvig!

**Pull request merged**